pkg/cloudprovider/providers/cloudstack/metadata.go

```diff
@@ -58,10 +59,20 @@ func (m *metadata) NodeAddresses(ctx context.Context, name types.NodeName) ([]v1
 		return nil, fmt.Errorf("could not get internal IP: %v", err)
 	}
 
-	return []v1.NodeAddress{
+	addresses := []v1.NodeAddress{
 		{Type: v1.NodeExternalIP, Address: externalIP},
 		{Type: v1.NodeInternalIP, Address: internalIP},
-	}, nil
+	}
+
+	hostname, err := m.get(metadataTypeHostname)
+	if err != nil {
+		return nil, fmt.Errorf("could not get hostname: %v", err)
+	}
+	if len(hostname) > 0 {
```

Comment:
Think it’s better (for readability) to check if it isn’t an empty string, instead of a non zero length.

This also us more consistent with other simular checks in this cloud provider.

Reply:
done